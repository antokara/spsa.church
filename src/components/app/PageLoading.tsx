import { Box, CircularProgress, Fade } from '@material-ui/core';
import * as React from 'react';

type TProps = {
  visible?: boolean;
};

/**
 * a page loading spinner that covers the whole page
 * with absolute left/top positioning and fade in/out capabilities
 */
const PageLoading: (props: TProps) => JSX.Element = ({
  visible = true
}: TProps): JSX.Element => (
  <Fade in={visible} data-testid="fade">
    <Box
      display="flex"
      justifyContent="center"
      alignItems="center"
      position="absolute"
      top={0}
      left={0}
      width="100%"
      height="100%"
      bgcolor="background.paper"
    >
      <CircularProgress data-testid="circular-progress" />
    </Box>
  </Fade>
);

export { PageLoading };
