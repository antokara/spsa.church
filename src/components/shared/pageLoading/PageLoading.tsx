import { Box, CircularProgress, Fade } from '@material-ui/core';
import * as React from 'react';

type TProps = {
  visible?: boolean;
  position?: string;
};

/**
 * a page loading spinner that covers the whole page
 * with absolute left/top positioning and fade in/out capabilities
 */
const PageLoading: (props: TProps) => JSX.Element = ({
  visible = true,
  position = 'absolute'
}: TProps): JSX.Element => (
  <Fade in={visible} data-testid="fade" unmountOnExit={true}>
    <Box
      display="flex"
      justifyContent="center"
      alignItems="center"
      position={position}
      top={0}
      left={0}
      width="100%"
      height="100%"
      minHeight="6rem"
      bgcolor="background.paper"
    >
      <CircularProgress data-testid="circular-progress" />
    </Box>
  </Fade>
);

export { PageLoading };
