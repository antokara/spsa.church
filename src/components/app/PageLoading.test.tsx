import { render, RenderResult } from '@testing-library/react';
import * as React from 'react';
import { PageLoading } from './PageLoading';

describe('PageLoading component', () => {
  let rr: RenderResult;
  let node: ChildNode | null;

  describe('with required props', () => {
    beforeEach(() => {
      rr = render(<PageLoading />);
      node = rr.container.firstChild;
    });

    it('renders the component', () => {
      expect(node).toMatchSnapshot();
    });

    describe('fade component', () => {
      let c: ChildNode | null;
      beforeEach(() => {
        c = rr.queryByTestId('fade');
      });

      it('gets rendered', () => {
        expect(c).toBeTruthy();
      });

      it('has opacity:1', () => {
        expect(c).toHaveStyle('opacity: 1');
      });
    });

    it('renders the circular progress', () => {
      expect(rr.queryByTestId('circular-progress')).toBeTruthy();
    });
  });

  describe('with prop.visible:true', () => {
    beforeEach(() => {
      rr = render(<PageLoading visible={true} />);
      node = rr.container.firstChild;
    });

    it('renders the component', () => {
      expect(node).toMatchSnapshot();
    });

    describe('fade component', () => {
      let c: ChildNode | null;
      beforeEach(() => {
        c = rr.queryByTestId('fade');
      });

      it('gets rendered', () => {
        expect(c).toBeTruthy();
      });

      it('has opacity:1', () => {
        expect(c).toHaveStyle('opacity: 1');
      });
    });

    it('renders the circular progress', () => {
      expect(rr.queryByTestId('circular-progress')).toBeTruthy();
    });

    describe('change prop.visible:false', () => {
      beforeEach(() => {
        rr.rerender(<PageLoading visible={false} />);
        node = rr.container.firstChild;
      });

      it('renders the component', () => {
        expect(node).toMatchSnapshot();
      });

      describe('fade component', () => {
        let c: ChildNode | null;
        beforeEach(() => {
          c = rr.queryByTestId('fade');
        });

        it('gets rendered', () => {
          expect(c).toBeTruthy();
        });

        it('has opacity:0', () => {
          expect(c).toHaveStyle('opacity: 0');
        });
      });
    });
  });
});
