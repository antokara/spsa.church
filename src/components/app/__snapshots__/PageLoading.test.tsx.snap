// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoading component with prop.visible:true change prop.visible:false renders the component 1`] = `
<div
  class="MuiBox-root MuiBox-root-58"
  data-testid="fade"
  style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    data-testid="circular-progress"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;

exports[`PageLoading component with prop.visible:true renders the component 1`] = `
<div
  class="MuiBox-root MuiBox-root-2"
  data-testid="fade"
  style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    data-testid="circular-progress"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;
